src/schema/api/Column/BaseColumn.h

```diff
@@ -0,0 +1,46 @@
+#ifndef LOODB_BASECOLUMN_H
+#define LOODB_BASECOLUMN_H
+
+#include "Column.h"
+#include "../Entry.h"
+#include "../../../filesystem/pagination/page/api/Page.h"
+#include "../../../filesystem/pagination/interface/api/Pager.h"
+#include <memory>
+#include <string>
+
```

Comment:
The `BaseColumnImpl` struct being here is ok here for now but it would be better to put it as a private inner class to prevent external access along with encapsulation principles

Reply:
Should all *Impl structs be private inner structs / classes? Is Impl as a class better than a struct? If so, why?

Reply:
Resolved: make pImpl structs/classes nested private.